src/HttpRequest.cc

```diff
@@ -668,6 +668,47 @@ HttpRequest::canHandle1xx() const
     return true;
 }
 
+bool
+HttpRequest::canUseContentLength() const
+{
+    // RFC 7230 Section 3.3.3 #4:
+    // conflicting Content-Length(s) mean a message framing error
+    if (header.conflictingContentLength())
+        return false;
+
+    // HTTP/1.0 requirements differ from HTTP/1.1
+    if (http_ver <= Http::ProtocolVersion(1,0)) {
+        const auto m = method.id();
+
+        // RFC 1945 section 8.3:
+        // "
+        //   A valid Content-Length is required on all HTTP/1.0 POST requests.
+        // "
+        if (m == Http::METHOD_POST)
+            return (content_length >= 0);
+
+        // RFC 1945 section 7.2:
+        // "
+        //   An entity body is included with a request message only when the
+        //   request method calls for one.
+        // "
+        // GET and HEAD do not define ('call for') an entity
+        if (m == Http::METHOD_GET || m == Http::METHOD_HEAD)
+            return (content_length < 0);
```

Comment:
In HTTP Content-Length:0 means there *is* an entity present. It just has 0 bytes of length. This distinction of that vs no entity (content_length < 0) is relevant for requests where the payload is a file or nested message etc. as happens in some HTTP extension specifications. Since this code is implementing the specification "only no-entity is valid" the condition is correct.